subsys/mpsl/fem/mpsl_fem_host.c

```diff
@@ -0,0 +1,144 @@
+
+	fem_pin_num_correction(&rx_en_pin, DT_GPIO_LABEL(DT_NODELABEL(nrf_radio_fem), rx_en_gpios));
+	nrf_gpio_pin_mcu_select(rx_en_pin, NRF_GPIO_PIN_MCUSEL_NETWORK);
+#endif
+
+#if DT_NODE_HAS_PROP(DT_NODELABEL(nrf_radio_fem), pdn_gpios)
+	uint8_t pdn_pin = DT_GPIO_PIN(DT_NODELABEL(nrf_radio_fem), pdn_gpios);
+
+	fem_pin_num_correction(&pdn_pin, DT_GPIO_LABEL(DT_NODELABEL(nrf_radio_fem), pdn_gpios));
+	nrf_gpio_pin_mcu_select(pdn_pin, NRF_GPIO_PIN_MCUSEL_NETWORK);
+#endif
+
+#if DT_NODE_HAS_PROP(DT_NODELABEL(nrf_radio_fem), mode_gpios)
+	uint8_t mode_pin = DT_GPIO_PIN(DT_NODELABEL(nrf_radio_fem), mode_gpios);
+
+	fem_pin_num_correction(&mode_pin, DT_GPIO_LABEL(DT_NODELABEL(nrf_radio_fem), mode_gpios));
+	nrf_gpio_pin_mcu_select(mode_pin, NRF_GPIO_PIN_MCUSEL_NETWORK);
+#endif
+
+#if DT_NODE_HAS_PROP(DT_NODELABEL(nrf_radio_fem), ant_sel_gpios)
+	uint8_t ant_sel_pin = DT_GPIO_PIN(DT_NODELABEL(nrf_radio_fem), ant_sel_gpios);
+
+	fem_pin_num_correction(&ant_sel_pin,
+			       DT_GPIO_LABEL(DT_NODELABEL(nrf_radio_fem), ant_sel_gpios));
+	nrf_gpio_pin_mcu_select(ant_sel_pin, NRF_GPIO_PIN_MCUSEL_NETWORK);
+#endif
+
+#if DT_NODE_HAS_STATUS(MPSL_FEM_SPI_IF, okay)
+	uint8_t cs_pin = DT_SPI_DEV_CS_GPIOS_PIN(MPSL_FEM_SPI_IF);
+	uint8_t sck_pin = DT_PROP(DT_BUS(DT_NODELABEL(nrf_radio_fem_spi)), sck_pin);
+	uint8_t miso_pin = DT_PROP(DT_BUS(DT_NODELABEL(nrf_radio_fem_spi)), miso_pin);
+	uint8_t mosi_pin = DT_PROP(DT_BUS(DT_NODELABEL(nrf_radio_fem_spi)), mosi_pin);
+
+	fem_pin_num_correction(&cs_pin, DT_SPI_DEV_CS_GPIOS_LABEL(MPSL_FEM_SPI_IF));
+	nrf_gpio_pin_mcu_select(cs_pin, NRF_GPIO_PIN_MCUSEL_NETWORK);
+	nrf_gpio_pin_mcu_select(sck_pin, NRF_GPIO_PIN_MCUSEL_NETWORK);
+	nrf_gpio_pin_mcu_select(miso_pin, NRF_GPIO_PIN_MCUSEL_NETWORK);
+	nrf_gpio_pin_mcu_select(mosi_pin, NRF_GPIO_PIN_MCUSEL_NETWORK);
+#endif
+
+	return 0;
+}
+#endif
+
+#if IS_ENABLED(CONFIG_MPSL_FEM_SIMPLE_GPIO)
+static int fem_simple_gpio_pins_forward(void)
+{
+#if !DT_NODE_EXISTS(DT_NODELABEL(nrf_radio_fem))
+#error Node with label 'nrf_radio_fem' not found in the devicetree.
+#endif
+
+#if DT_NODE_HAS_PROP(DT_NODELABEL(nrf_radio_fem), ctx_gpios)
+	uint8_t ctx_pin = DT_GPIO_PIN(DT_NODELABEL(nrf_radio_fem), ctx_gpios);
+
+	fem_pin_num_correction(&ctx_pin, DT_GPIO_LABEL(DT_NODELABEL(nrf_radio_fem), ctx_gpios));
+	nrf_gpio_pin_mcu_select(ctx_pin, NRF_GPIO_PIN_MCUSEL_NETWORK);
+#endif
+
+#if DT_NODE_HAS_PROP(DT_NODELABEL(nrf_radio_fem), crx_gpios)
+	uint8_t crx_pin = DT_GPIO_PIN(DT_NODELABEL(nrf_radio_fem), crx_gpios);
+
+	fem_pin_num_correction(&crx_pin, DT_GPIO_LABEL(DT_NODELABEL(nrf_radio_fem), crx_gpios));
+	nrf_gpio_pin_mcu_select(crx_pin, NRF_GPIO_PIN_MCUSEL_NETWORK);
+#endif
+
+	return 0;
+}
+#endif
+
+static int mpsl_fem_host_init(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	int err = 0;
+
+#if IS_ENABLED(CONFIG_MPSL_FEM_NRF21540_GPIO)
+	err = fem_nrf21540_gpio_pins_forward();
+#elif IS_ENABLED(CONFIG_MPSL_FEM_SIMPLE_GPIO)
+	err = fem_simple_gpio_pins_forward();
+#else
```

Comment:
nit:
```
if (IS_ENABLED(CONFIG_MPSL_FEM_NRF21540_GPIO)) {
    err = fem_nrf21540_gpio_pins_forward();
} else if (IS_ENABLED(CONFIG_MPSL_FEM_SIMPLE_GPIO)) {
    err = fem_simple_gpio_pins_forward();
}
```
This has the general advantage that the compiler will report build errors (if there are any) inside sections guarded by `CONFIG_MPSL_FEM_XYZ` configs, even if the configs were set to false.

Reply:
I implemented your suggestion and replaced `#error` with an `__ASSERT`. Please check if it's OK now.